backstory/src/main/java/com/google/sps/story/DatamuseRequestClient.java

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.story.data;
+
+import com.google.sps.APINotAvailableException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * A class to make requests of the Datamuse API.
+ */
+public class DatamuseRequestClient {
+  /** the url to access the Datamuse database we're querying */
+  private static final String DATAMUSE_URL = "http://api.datamuse.com/words?";
+
+  /** holds the base url to query */
+  private final String queryUrl;
+
+  /**
+   * Constructs a request client with the default Datamuse url.
+   */
+  public DatamuseRequestClient() {
+    this(DATAMUSE_URL);
+  }
+
+  /**
+   * Constructs a request client that will query
+   * the given URL (for testing purposes)
+   *
+   * @param queryUrl the url to instantiate this request client with
+   */
+  public DatamuseRequestClient(String queryUrl) {
+    this.queryUrl = queryUrl;
+  }
+
+  /**
+   * Gets an array of adjectives related to the passed-in noun, which will have
+   * a passed-in cap on its size (array returned might be smaller if database
+   * does not have enough related adjectives). Fetches these adjectives
+   * by querying the Datamuse database.
+   *
+   * @param noun the noun to get adjectives related to (must be one word)
+   * @param cap the maximum number of adjectives to retrieve
+   * @return an array of adjectives related to the noun
+   * @throws IllegalArgumentException if noun is more than one word (has a space in it)
+   * @throws APINotAvailableException if Datamuse API cannot be reached
+   * @throws RuntimeException if JSON received back from the API cannot be parsed (JSONException)
+   *    or does not have objects of type JSON (ClassCastException)
+   */
+  public String[] getRelatedAdjectives(String noun, int cap)
+      throws IllegalArgumentException, APINotAvailableException, RuntimeException {
+    if (noun.contains(" ")) {
```

Comment:
Do you want to also check for other whitespace characters?
https://stackoverflow.com/a/4067825

Reply:
I'm not sure if it's necessary, but I'm happy to make the change just to be safe.

Reply:
Done